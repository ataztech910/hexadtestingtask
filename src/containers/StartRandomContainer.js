import { connect } from 'react-redux';
import StartRandomRating from '../components/startRandomRating';
import { startRandomProcess } from '../actions/ratings';

const mapDispatchToProps = dispatch => ({
  startRandomProcess: rating => dispatch(startRandomProcess(rating)),
});

const mapStateToProps = (state) => {
  return { state };
};


export default connect(
  mapStateToProps, mapDispatchToProps,
)(StartRandomRating);
