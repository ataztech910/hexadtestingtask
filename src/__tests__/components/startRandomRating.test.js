import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import StartRandomRating from '../../components/startRandomRating';
import StartRandomContainer from '../../containers/StartRandomContainer';

configure({ adapter: new Adapter() });
const dispatch = jest.fn();

const mockStore = configureMockStore();
const store = mockStore({});

describe('Start random process button check', () => {
  const props = {
    startRandomProcess: dispatch,
    state: {
      startProcess: false,
    },
  };
  const element = shallow(<StartRandomRating {...props} />);
  const elementContainer = shallow(<StartRandomContainer store={store} />).dive();
  it('renders properly', () => {
    expect(element).toMatchSnapshot();
  });
  it('Container renders properly', () => {
    expect(elementContainer).toMatchSnapshot();
  });

  it('Random rating process on click', () => {
    element.find('.randomRatingProcess').simulate('click', { preventDefault() { } });
    expect(dispatch.mock.calls.length).toEqual(1);
  });
});
