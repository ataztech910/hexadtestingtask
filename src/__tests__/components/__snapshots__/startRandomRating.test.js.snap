// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start random process button check Container renders properly 1`] = `ShallowWrapper {}`;

exports[`Start random process button check renders properly 1`] = `ShallowWrapper {}`;
